cmd/maya-apiserver/cstor-operator/cspc/handler.go

```diff
@@ -122,26 +122,41 @@ func (c *Controller) syncCSPC(cspc *apis.CStorPoolCluster) error {
 	//}
 	openebsNameSpace := env.Get(env.OpenEBSNamespace)
 	if openebsNameSpace == "" {
+
+		message := fmt.Sprint("Could not sync CSPC: got empty namespace for openebs from env variable")
+		c.recorder.Event(cspc, apis.EventTypeError, "Getting Namespace", message)
 		return errors.Errorf("Could not sync CSPC {%s}: got empty namespace for openebs from env variable", cspc.Name)
```

Comment:
returning 'nil' is better to avoid requeueing

Reply:
my take on this is: returning nil on all the cases except when we know that faster reconcilation helps.
one such example is: if BDCs are just created, I expect faster reconcilation (may be with induced delay like for a second)

Reply:
Ok. Will fix it.

Reply:
What should be the ideal behavior if `OPENEBS_NAMESPACE` is not set? We are setting `OPENEBS_NAMESPACE` through operator only. If it is not set then there is a problem in deployment. So we should not start Maya-apiserver if it is not set. Or we should set from `maya-apiserver` itself?